gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
@hasLeland Is there actually other case outside of the null case? Presumably any repeated type is going to be `null` or `[1,..,n]`, yeah?
```
		if len(v) > 0 && !(v[0] == '[' || v[len(v)-1] == ']') && v != "null" {
			v = "[" + v + "]"
		}
```
Definitely makes the resulting if a tad gnarlier, _probably_ still cheaper than multiple decodes. Particularly if we suspect `csv` style to be a common case, failing throw for each one can really add up. I'm curious what @hasIan has to say though. 

https://play.golang.org/p/y9CNqU18-D